addons/survey/models/survey_user_input.py

```diff
@@ -782,3 +782,32 @@ def _get_answer_score_values(self, vals, compute_speed_score=True):
             'answer_is_correct': answer_is_correct,
             'answer_score': answer_score
         }
+
+    def _get_answer_value_field(self):
+        if self.answer_type == 'char_box':
+            return 'value_char_box'
+        elif self.answer_type == 'text_box':
+            return 'value_text_box'
+        elif self.answer_type == 'numerical_box':
+            return 'value_numerical_box'
+        elif self.answer_type == 'date':
+            return 'value_date'
+        elif self.answer_type == 'datetime':
+            return 'value_datetime'
+        elif self.answer_type == 'suggestion':
+            return 'suggested_answer_id.value'
+
+    def _get_matching_answer_domain(self):
```

Comment:
Would name it "_get_answer_matching_domain" for consistency with other methods (could also order the method alphabetically inside private methods, so probably before "_get_answer_score_values")